ktor-server/ktor-server-plugins/ktor-server-request-validation/jvmAndNix/src/io/ktor/server/plugins/requestvalidation/RequestValidation.kt

```diff
@@ -72,6 +76,19 @@ public val RequestValidation: RouteScopedPlugin<RequestValidationConfig> = creat
             throw RequestValidationException(content, failures.flatMap { it.reasons })
         }
     }
+
+    if (!pluginConfig.validateContentLength) return@createRouteScopedPlugin
+
+    on(ReceiveRequestBytes) { call, body ->
+        val contentLength = call.request.contentLength() ?: return@on body
+
+        return@on application.writer {
+            val count = body.copyTo(channel)
+            if (count != contentLength) {
+                throw IOException("Content length mismatch. Actual $count, expected $contentLength.")
```

Comment:
maybe it's better to throw `ValidationException` here

Reply:
I would like to keep the IO exception here: because engines already perform validation internally with IOException and it will be backward compatible